src/renderer/wg_engine/tvgWgRenderer.cpp

```diff
@@ -95,10 +237,206 @@ bool WgRenderer::clear() {
+                        // draw to stencil
+                        mPipelineEmpty.set(renderPassEncoder);
+                        mPipelineBindGroupEmpty.bind(renderPassEncoder, 0);
+                        renderData->mGeometryDataFill[j]->draw(renderPassEncoder);
+
+                        // draw brush
+                        renderData->mPipelineBase->set(renderPassEncoder);
+                        renderData->mPipelineBindGroup->bind(renderPassEncoder, 0);
+                        mGeometryDataPipeline.draw(renderPassEncoder);
+                    }
+                }
+            }
+            // end render pass
+            wgpuRenderPassEncoderEnd(renderPassEncoder);
+            wgpuRenderPassEncoderRelease(renderPassEncoder);
+
+            // release backbuffer and present
+            wgpuTextureViewRelease(backBufferView);
+        }
+        commandsBuffer = wgpuCommandEncoderFinish(commandEncoder, &commandBufferDesc);
+        wgpuCommandEncoderRelease(commandEncoder);
+    }
+    // queue submit and command buffer release
+    wgpuQueueSubmit(mQueue, 1, &commandsBuffer);
+    wgpuCommandBufferRelease(commandsBuffer);
+    
+    // go to the next frame
+    wgpuSwapChainPresent(mSwapChain);
+
+    // clear current render datas
+    mRenderDatas.clear();
     return true;
 }
 
+bool WgRenderer::target(uint32_t* buffer, uint32_t stride, uint32_t w, uint32_t h) {
+    // check properties
+    assert(w > 0);
+    assert(h > 0);
+
+    // store target surface properties
+    mTargetSurface.stride = stride;
+    mTargetSurface.w = w;
+    mTargetSurface.h = h;
+
+    // update view matrix
+    mViewMatrix[ 0] = +2.0f / w; mViewMatrix[ 1] = +0.0f;     mViewMatrix[ 2] = +0.0f; mViewMatrix[ 3] = +0.0f;
+    mViewMatrix[ 4] = +0.0f;     mViewMatrix[ 5] = -2.0f / h; mViewMatrix[ 6] = +0.0f; mViewMatrix[ 7] = +0.0f;
+    mViewMatrix[ 8] = +0.0f;     mViewMatrix[ 9] = +0.0f;     mViewMatrix[10] = -1.0f; mViewMatrix[11] = +0.0f;
+    mViewMatrix[12] = -1.0f;     mViewMatrix[13] = +1.0f;     mViewMatrix[14] = +0.0f; mViewMatrix[15] = +1.0f;
+
+    // return result
+    return true;
+}
+
+// target for native window handle
 bool WgRenderer::target(void* window, uint32_t w, uint32_t h) {
+    // check properties
+    assert(window);
+    assert(w > 0);
+    assert(h > 0);
+
+    // store target surface properties
+    mTargetSurface.stride = w;
+    mTargetSurface.w = w > 0 ? w : 1;
+    mTargetSurface.h = h > 0 ? h : 1;
+
+    // update view matrix
+    mViewMatrix[ 0] = +2.0f / w; mViewMatrix[ 1] = +0.0f;     mViewMatrix[ 2] = +0.0f; mViewMatrix[ 3] = +0.0f;
+    mViewMatrix[ 4] = +0.0f;     mViewMatrix[ 5] = -2.0f / h; mViewMatrix[ 6] = +0.0f; mViewMatrix[ 7] = +0.0f;
+    mViewMatrix[ 8] = +0.0f;     mViewMatrix[ 9] = +0.0f;     mViewMatrix[10] = -1.0f; mViewMatrix[11] = +0.0f;
+    mViewMatrix[12] = -1.0f;     mViewMatrix[13] = +1.0f;     mViewMatrix[14] = +0.0f; mViewMatrix[15] = +1.0f;
+
+    #ifdef _WIN32
```

Comment:
So far, this code won't function on any other platforms. Our `wg_engine` should not rely on platform-specific code unless there are valid reasons to do so. Please add comments and exceptions (with reasons) using TODO or FIXME comments so that we can revisit this temporary code later.